### ISSUE TITLE
calloc.free() doen't throw an ArgumentError if memory cannot be free

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/ffi#115]
Accordint to the 'Allocator.free()' [documentation](https://api.dart.dev/dev/2.14.0-362.0.dev/dart-ffi/Allocator/free.html) 

> Throws an ArgumentError if the memory pointed to by pointer cannot be freed.

But in case if the memory pointed by pointer was already freed, then no `ArgumentError` is thrown. For example

```dart
import "dart:ffi";
import "package:ffi/ffi.dart";

void main() {
  Pointer<IntPtr> p1 = calloc<IntPtr>(2);
  Pointer<IntPtr> p2 = new Pointer.fromAddress(p1.address + sizeOf<IntPtr>());
  try {
    p1.value = 1;
    p2.value = 42;
    print(p1[0]);
    print(p1[1]);
  } finally {
    calloc.free(p1);  // this frees memory pointed by p2 as well
    calloc.free(p2);  // There must be ArgumentError here
  }
}
```
This program produces:
```
1
42

Process finished with exit code -1073740940 (0xC0000374)
```
No expected `ArgumentError`

Tested on `Dart SDK version: 2.14.0-363.0.dev (dev) (Wed Jul 28 23:43:25 2021 -0700) on "windows_x64" `

**Comment by dcharkes** [comment]
The documentation promises too much, we should update the documentation.

On windows we can report errors, but in unix systems we cannot: `free` returns void in C. https://github.com/dart-lang/ffi/blob/db710989e4f9904cfc1c784f25e72030d7098151/lib/src/allocation.dart#L145-L166

**Comment by timsneath** [comment]
Fixed with #115
